### ISSUE TITLE
Brackets for Linux

### ISSUE THREAD

**Issue by benlumia007** [adobe/brackets#14667]
Is it possible to update so that brackets can work with linux due to libcurl3 and libcurl4. I know that i can recompile by replacing libcurl4 but i would like a minor fixed

**Comment by ryuwiz** [comment]
yeah, when I installing brackets it need to remove virtualbox because different on libcurl.

**Comment by krizalidzero** [comment]
When I recompile it to work with libcurl4, the program works, but the text gets fuzzy and basically becomes unusable,

**Comment by aniketbarik** [comment]
> When I recompile it to work with libcurl4, the program works, but the text gets fuzzy and basically becomes unusable,

I have this issue too. The fuzzy text makes even compiling it pointless because fuzzy text renders it useless. I wish Brackets developers would fix this issue with libcurl. Last Brackets update was in June, 2018.

**Comment by Taariqq** [comment]
There was an extension for linux to handle just that, no?

**Comment by shubhsnov** [comment]
@benlumia007 @krizalidzero @aniketbarik Can you guys help with some more specific info of exactly what is not working?

Is the problem with the installation of the Brackets package on Linux or is the problem with building from source?
Also please specify the flavour/OS you guys are on. 
Thanks for reporting this.

**Comment by benlumia007** [comment]
@shubhsnov 

since most of people here are using linux of some kind of distro. The current version of Brackets is still using libcurl3 which is one of the dependency needed

In this case, some of us have virtualbox installed, and i think it started at the last vrsion 5.2.28 or early 6.0 have moved to libcurl4 as the dependency. 

if brackets can replace libcurl3 with libcurl4, we as linux users should be able to install Brackets.

I hope that make sense.

**Comment by krizalidzero** [comment]
> 
> 
> @benlumia007 @krizalidzero @aniketbarik Can you guys help with some more specific info of exactly what is not working?
> 
> Is the problem with the installation of the Brackets package on Linux or is the problem with building from source?
> Also please specify the flavour/OS you guys are on.
> Thanks for reporting this.

Hello. Thank you for asking.
I was using Ubuntu Mate 18.10 at the time of the report.

Brackets can't be installed via the deb package unless you modify it to require libcurl4 (it normally requires version 3). And when you modify it, it runs fine, except the text looks extremely fuzzy.

I was using it at work, and I don't have access to that computer anymore to make a screenshot or similar.

**Comment by benlumia007** [comment]
is the .deb package for Brackets can't be install if virtualbox is install becuse virtualbox and some new applications uses libcurl4, brackets currently have libcurl3, if i continue to install brackets, it will then automatically uninstall VirtualBox. 

the point is, Brackets needs libcurl4 as a dependency package for .deb packages.

**Comment by shubhsnov** [comment]
@benlumia007 @krizalidzero @aniketbarik I've been investigating this issue, and it seems that the shell actually doesn't even need the libcurl dependency. The dependency was added here by [replicating](https://github.com/adobe/brackets-shell/pull/305) the dependencies for Google Chrome 28.

But chromium itself has moved on from that dependency as can be seen [here](https://chromium.googlesource.com/chromium/src/+/c95584d4c062b6034f580798bab02dcdc588c141).

I built the debian package after removing the libcurl3 dependency and was able to use Brackets and VirtualBox side by side without any issue. I didn't notice any fuzzy text as well.

We're working on bringing this change into the actual build. Meanwhile, you guys can also help us by creating a package without the libcurl dependency and see if you run into any other functional issues on installation and usage.

Thanks again for all the valuable info.

**Comment by krizalidzero** [comment]
Alright, I'm trying again in Ubuntu 19.04, it does work without libcurl4 (as in it installs fine and runs fine), but the fuzzy text is still there. Might be a platform issue.

![Captura de pantalla -2019-04-25 20-24-04](https://user-images.githubusercontent.com/44400318/56732405-2ed65880-6798-11e9-9812-59875cda2179.png)

The screenshot is in Spanish, but I have used the OS in Spanish, Japanese and English with the same results.

![Captura de pantalla -2019-04-25 20-26-06](https://user-images.githubusercontent.com/44400318/56732457-5fb68d80-6798-11e9-88b4-d9b26c61b55c.png)


But I'm pretty sure the problem is Ubuntu, since this also happened to me with NetBeans 10 and other editors.

**Comment by shubhsnov** [comment]
@krizalidzero Thanks for verifying. Great to know that it works for you now. I think the fuzzy text is not related to the libcurl3 issue. You should create a separate issue for that. 
 
Did you ever try installing with libcurl3, and did it solve the fuzzy text problem?

**Comment by shubhsnov** [comment]
@krizalidzero The font issue seems to be totally independent and might be related to [this](https://github.com/adobe/brackets/issues/14290#issuecomment-388323987). You can take a look at @gautam0217's solution and see if it works for you.

**Comment by krizalidzero** [comment]
Hello and thanks a lot for the help.
I actually installed KDE Plasma and the problem of fuzzy text isn't happening in a KDE environment, so it's all good.

Thanks again!

**Comment by benlumia007** [comment]
This topic wasn't meant for blurry/fuzzy fonts, this topic was meant to be due to  the libcurl4 for newer version. why was this not solved before Brackets 1.14 was released. 

I don't think Adobe is taking this seriously enough to fix this problem.

**Comment by shubhsnov** [comment]
We did try, but there were time constraints. We really wanted to put out a release as soon as possible as well as provide some value. Upon some investigation, it turned out that Brackets which uses CEF as a shell doesn't exactly need libcurl3 let alone libcurl4. The libcurl3 dependency was removed in CEF 2840 (Chromium based) while we use 2704 so we just didn't take the risk even though it very clearly is a redundant dependency. 

https://codereview.chromium.org/2271063004/diff/1/chrome/installer/linux/debian/build.sh

Be assured that we have prioritized this issue and are working to update the CEF for the LINUX at this very moment. Please bear with us. 

https://github.com/adobe/brackets-shell/pull/666

**Comment by aniketbarik** [comment]
> @benlumia007 @krizalidzero @aniketbarik Can you guys help with some more specific info of exactly what is not working?
> 
> Is the problem with the installation of the Brackets package on Linux or is the problem with building from source?
> Also please specify the flavour/OS you guys are on.
> Thanks for reporting this.

Sorry, for the late reply. I'm on Ubuntu 18.04.2 with GNOME DE.
If I try installing Brackets v1.14 without compiling and changing the dependency to libcurl4, it says that it needs to remove Clamav, Clamtk, curl, libcurl4, and a couple more packages. Changing the dependency to libcurl4 fixes this.

> Be assured that we have prioritized this issue and are working to update CEF for the Linux at this very moment. Post which we'll update Brackets dependencies and remove the redundant dependencies altogether. Please bear with us.

Thanks. That's good to know. Cheers.

**Comment by Ray6464** [comment]
There is a duplicate issue #14667

**Comment by benlumia007** [comment]
I still seem to understand why this wasn't implemented yet to be fixed.

**Comment by Taariqq** [comment]
> I still don't seem to understand why this wasn't implemented yet to be fixed.

Perhaps because Brackets is dead ... again!

**Comment by benlumia007** [comment]
it sounds like it, i mean, everyone moved to visual studio code

**Comment by Taariqq** [comment]
> it sounds like it, i mean, everyone moved to visual studio code

... and from VSCode, I moved to Apache NetBeans. NetBeans is great for PHP development out of the box whereas vscode, one has to configure a bit. Plus, I like the workflow in NetBeans, coupled with the Dracula theme, everything works seamlessly.

**Comment by Taariqq** [comment]
The 'live preview' in vscode is not that good. And 'live reload' does not want to work with Chrome and vscode directly ... so I have have Brackets running with 'live reload' extension on and then I _can_ activate live reload in Chrome.
It is not as responsive as Brackets, plus it only works on save, but it is going to have to do for now.

VSCode people are downright arrogant. They just won't implement live preview.

**Comment by raphaelcasimir** [comment]
Here is a link (maybe temporary) to the repackaged deb that works on linux (It is possible to verify that I only touched the control file via unzipping and md5ing). 2020.10.02 version.
http://c.raph.io/index.php/s/9s86JB9RqPwgqfN

**Comment by benlumia007** [comment]
I believed that the package only removes and adds libcurl4, but make sure if u do this way, u don't do updates on it, if brackets have a new version, you're probably be stuck using the repackaging one. or else for every new release, it will need to be repackaged without that libcurl3.

**Comment by GogoAkira** [comment]
Apparently libcurl is not needed anyway per shubhsnov, he wrote above, so removing the libcurl3 works for me without the need to add libcurl4 as dependency.

**Comment by benlumia007** [comment]
the problem you can remove it by yourself, but i rather have adobe brackets remove the libcurl altogether. being able o update to latest version is needed.

**Comment by GogoAkira** [comment]
In case someone has libcrul4 as dependency if they uninstall brackets, will that uninstall the libcurl4 also? I tried to understand the other problems you mentioned before, I'm not a developer really, just starting out :)
Yes that's true you have to do it manually, but I found debtool (a script) that kind of makes is easy to repackage, takes like few seconds :)
